### DISCUSSION TITLE
Is it possible to have a "full" nightly build?

### DISCUSSION THREAD

**Discussion by hariso** [goreleaser/goreleaser#2869] [Ideas, Issue Triage, and General discussions]
Hello!

Firstly, thank you for this great tool! 

I'm looking for a way to provide "full" nightly builds for a project, i.e.
* have it scheduled on a predefined time (e.g. once per day)
* have a new GitHub release (with all the usual assets) 
* a new Docker image (optional)
* clean up older releases after some time

I found the Nightly build feature in GoReleaser Pro. However, there it states that GitHub releases are skipped. 

Hence, my question is: is it possible to achieve what I want (using Pro)?

Thanks in advance!

Haris

**Comment by caarlos0** [comment]
Hey, its not possible with the nightly feature.

What you want seems more likely "alpha" versions or something like that... in which case you can just tag the release normally and use templates to adjust things in the way you need them (e.g. docker image names, wether it should be a prerelease, etc)

For that you don't even need goreleaser pro :)

The nightly feature intent is to be used to build every commit and push the artifacts/docker images somewhere. I should come up with better docs/examples of that at some point.

Hope this helps!

**Reply by hariso** [reply]
@caarlos0 thanks for the reply! I've added one more thing to the list, which I think makes it distinct from alpha releases, and that is that I'd like it to be scheduled on a daily basis and not run for every commit.

At the project I'm working on we already have a setup where pushing a tag triggers a workflow which, using GoReleaser, builds the artifacts and docker images etc.

What we currently have in plan for the nightly builds is to tag the main branch once in a day (night :) ), and then let GoReleaser do its job. It makes the workflow a bit more complex and requires cleanup of older releases and images, which is why wanted to check if the Pro version's nightly build can help with that.

**Reply by caarlos0** [reply]
> What we currently have in plan for the nightly builds is to tag the main branch once in a day (night :) ), and then let GoReleaser do its job. It makes the workflow a bit more complex and requires cleanup of older releases and images, which is why wanted to check if the Pro version's nightly build can help with that.

yep, that would be basically it...

cleaning up old releases is not something I have planned right now as well, should be relatively straightforward to do so with some shell script/github actions though.

A note that you might be already aware of: if you tag via actions, you'll need a PAT. Tags created using the action `GITHUB_TOKEN` do not trigger build et al - or at least this was true when I last checked.

**Reply by hariso** [reply]
Yes, you're right about the PATs and actions. And that thing with actions and PATs bit me actually. I spent some time wondering why isn't my thing getting triggered. And only after that I actually went to the docs.: ) 

I was a bit reluctant to do that, since the PATs to be manually refreshed (at least as far as I could find out).

And yes, just a couple of hours ago I went with a solution which combine some existing GH actions which clean up releases and Docker images. Now all what is left is to clean up the tags and that should be it. I also wanted to use "static" tags and force overwrites of tags and releases, but I felt it could end up being confusing for some folks.

Thank you very much for all the useful info you provided!

**Reply by caarlos0** [reply]
nice! Glad to see it worked out! 

Also, if you end up writing something about it - or want to write about it, its something i'd love to feature on https://blog.goreleaser.com :) 

Cheers!

**Reply by hariso** [reply]
I'll be happy to do so! We'll have this setup running for some time running in this project, so we can fix any bugs in it. After that I'll be free to reach out to you with a blog post.: )

**Reply by caarlos0** [reply]
amazing! Ping me anytime at gophers slack or [our discord ](https://discord.gg/RGEBtg8vQ6)
